From 9c4764224aded552fb855b1c2b85b26d2b894adf Mon Sep 17 00:00:00 2001
From: Alexander Tsoy <alexander@tsoy.me>
Date: Tue, 29 Oct 2019 18:17:51 +0300
Subject: [PATCH] Fix building against libglvnd-provided EGL headers

Add missing EGL/eglmesaext.h include. When building against mesa-provided
headers, this header is included indirectly via EGL/eglext.h.

Also fix typo COGL_EGL_INCLUDE -> COGL_EGL_INCLUDES
---
 configure.ac | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index da188edd..b0b0c94d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -670,10 +670,15 @@ AS_IF([test "x$enable_gles1" = "xyes"],
             AC_SUBST([COGL_EGL_INCLUDES])
 
             AC_CHECK_HEADERS([EGL/eglext.h],
-                             [COGL_EGL_INCLUDES="$COGL_EGL_INCLUDE
+                             [COGL_EGL_INCLUDES="$COGL_EGL_INCLUDES
 #include <EGL/eglext.h>"],
                              [],
                              [$COGL_EGL_INCLUDES])
+            AC_CHECK_HEADERS([EGL/eglmesaext.h],
+                             [COGL_EGL_INCLUDES="$COGL_EGL_INCLUDES
+#include <EGL/eglmesaext.h>"],
+                             [],
+                             [$COGL_EGL_INCLUDES])
 
             # Check for a GLES 1.x Common Profile library with/without EGL.
             #
@@ -1178,6 +1183,12 @@ AS_IF([test "x$NEED_EGL" = "xyes" && test "x$EGL_CHECKED" != "xyes"],
               [],
               [AC_MSG_ERROR([Unable to locate required EGL headers])],
               [#include <EGL/egl.h>])
+            AC_CHECK_HEADERS(
+              [EGL/eglmesaext.h],
+              [],
+              [AC_MSG_ERROR([Unable to locate required EGL headers])],
+              [#include <EGL/egl.h>
+#include <EGL/eglext.h>])
 
             AC_CHECK_LIB(EGL, [eglInitialize],
               [COGL_EXTRA_LDFLAGS="$COGL_EXTRA_LDFLAGS -lEGL"],
@@ -1188,7 +1199,8 @@ AS_IF([test "x$NEED_EGL" = "xyes" && test "x$EGL_CHECKED" != "xyes"],
           )
 
         COGL_EGL_INCLUDES="#include <EGL/egl.h>
-#include <EGL/eglext.h>"
+#include <EGL/eglext.h>
+#include <EGL/eglmesaext.h>"
         AC_SUBST([COGL_EGL_INCLUDES])
       ])
 
-- 
2.22.0

